File: regional/regional/2015/I_O/EggDrop/EggDrop.pl
$lin = <> ;
chomp $lin ;
($n, $k) = split " ", $lin ;
my $hi = $k - 1 ;
my $lo = 2 ;
while (<>) {
   $f = (split " ")[0] ;
   $hi = $f - 1 if /BROKEN/ && $f <= $hi ;
   $lo = $f + 1 if /SAFE/ && $f >= $lo ;
}
print "$lo $hi\n" ;

File: regional/PacificNW/2015/I_O/EggDrop/EggDrop.py
lin = raw_input().split(" ")
n = int(lin[0])
k = int(lin[1])
lo = 2
hi = k-1
for i in range(n):
   lin = raw_input().split(" ")
   if lin[1] == 'SAFE':
      lo = max(lo, int(lin[0])+1)
   if lin[1] == 'BROKEN':
      hi = min(hi, int(lin[0])-1)
print lo, hi
